### PR TITLE
support s3 sinergise source

### PR DESCRIPTION
**Pull request by philvarner**
- https://github.com/stactools-packages/sentinel2/issues/19
- #16 

### Changed

- TCI asset now has role `visual` instead of `data`
- Change platform and constellation to use best practices (lowercase, no spaces)
- Change asset names to use underscore instead of dash
- Band 8A corrected to have common name `nir08` instead of `rededge`

### Added

- Add support for AWS S3 Open Data format (produced by Sinergise)
- Band 9 now has common name nir09 and Band 10 has common name cirrus

### CODE REVIEW

File: .github/workflows/continuous-integration.yml [hunk 1]
```diff
@@ -25,7 +25,7 @@ jobs:
     runs-on: ubuntu-latest
     strategy:
       matrix:
-        python-version: ["3.7", "3.8", "3.9", "3.10"]
+        python-version: ["3.8", "3.9", "3.10"]
```
Comment (philvarner) [inline .github/workflows/continuous-integration.yml:28]: de-support python 3.7 (doesn't support Final type or walrus operator)

File: setup.cfg [hunk 2]
```diff
@@ -18,18 +18,20 @@ keywords =
...
 
 [options]
-python_requires = >=3.7
+python_requires = >=3.8
 package_dir =
     = src
 packages = find_namespace:
 install_requires =
     stactools >= 0.3.0
+    pystac
```
Comment (philvarner) [inline setup.cfg:32]: these were already transitive dependencies, so making them explicit.

File: src/stactools/sentinel2/constants.py [hunk 3]
```diff
@@ -71,17 +78,19 @@
                 full_width_half_max=0.145),
     'B8A':
     Band.create(name='B8A',
-                common_name='rededge',
+                common_name='nir08',
```
Comment (philvarner) [inline src/stactools/sentinel2/constants.py:81]: this was incorrect.

File: src/stactools/sentinel2/constants.py [hunk 4]
```diff
@@ -103,52 +112,75 @@
...
-    ],
+BANDS_TO_RESOLUTIONS: Final[Dict[str, List[int]]] = {
+    'B01': [60],
+    'B02': [10, 20, 60],
+    'B03': [10, 20, 60],
+    'B04': [10, 20, 60],
+    'B05': [20, 60],
+    'B06': [20, 60],
+    'B07': [20, 60],
+    'B08': [10, 20, 60],
```
Comment (philvarner) [inline src/stactools/sentinel2/constants.py:123]: B08 has 20 and 60m assets in the sinergise dataset

File: src/stactools/sentinel2/constants.py [hunk 5]
```diff
@@ -103,52 +112,75 @@
...
-    ],
-    'B12': [
-        20,
-        60,
-    ],
+    'B11': [20, 60],
+    'B12': [20, 60],
 }
+
+L2A_IMAGE_PATHS: Final[List[str]] = [
```
Comment (philvarner) [inline src/stactools/sentinel2/constants.py:131]: for the sinergise dataset, there's no metadata about what assets are available. We assume they all exist. Alternative is to have to do a list operation against S3 or to pass in all of the known assets from the S3 Inventory and only do the S3 query if all of the expected ones are not there.

File: src/stactools/sentinel2/granule_metadata.py [hunk 6]
```diff
@@ -21,6 +21,12 @@ def __init__(self,
 
         self._root = XmlElement.from_file(href, read_href_modifier)
 
+        tile_id = self._root.find_text('n1:General_Info/TILE_ID')
```
Comment (philvarner) [inline src/stactools/sentinel2/granule_metadata.py:24]: use the granule's tile id for this dataset.

File: src/stactools/sentinel2/granule_metadata.py [hunk 7]
```diff
@@ -167,8 +173,39 @@ def metadata_dict(self):
...
+        id_parts = self.product_id.split('_')
+
+        # Remove PDGS Processing Baseline number
+        id_parts = [part for part in id_parts if not part.startswith('N')]
+
+        return '_'.join(id_parts)
+
+    @property
+    def platform(self) -> Optional[str]:
+        if self.tile_id.startswith("S2A"):
```
Comment (philvarner) [inline src/stactools/sentinel2/granule_metadata.py:200]: derive the platform from the tile id

File: src/stactools/sentinel2/constants.py [hunk 8]
```diff
@@ -1,26 +1,33 @@
...
+        ProviderRole.PRODUCER, ProviderRole.PROCESSOR, ProviderRole.LICENSOR
+    ],
     url='https://earth.esa.int/web/guest/home')
 
-SAFE_MANIFEST_ASSET_KEY = "safe-manifest"
-INSPIRE_METADATA_ASSET_KEY = "inspire-metadata"
-PRODUCT_METADATA_ASSET_KEY = "product-metadata"
-GRANULE_METADATA_ASSET_KEY = "granule-metadata"
-DATASTRIP_METADATA_ASSET_KEY = "datastrip-metadata"
+SAFE_MANIFEST_ASSET_KEY: Final[str] = "safe_manifest"
```
Comment (philvarner) [inline src/stactools/sentinel2/constants.py:23]: use underscore instead of `-` for these, per common JSON recommendations

**Review by matthewhanson — APPROVED** [review]

**Pull request merged**